### PR TITLE
Add more tests to auth service APIs

### PR DESCRIPTION
**Pull request by arielshaqed**

### CODE REVIEW

**Comment by arielshaqed** [comment]
@itaiad200 sorry, I asked you to review my draft PR.  Now without `[DEBUG]` markers.

File: auth/service_test.go [hunk 1]
```diff
@@ -379,3 +442,185 @@ func TestDBAuthService_Authorize(t *testing.T) {
...
+			}
+			gotList, _, err := s.ListUsers(&model.PaginationParams{Amount: -1})
+			if err != nil {
+				t.Fatalf("ListUsers: %s", err)
+			}
+			gotUsers := make([]string, 0, len(testCase.userNames))
+			for _, user := range gotList {
+				gotUsers = append(gotUsers, user.DisplayName)
+			}
+			sort.Strings(gotUsers)
```
Comment (johnnyaug) [inline auth/service_test.go:478]: Is it a requirement that users are returned sorted? If so, removing this line would make the test stronger.

File: auth/service_test.go [hunk 2]
```diff
@@ -379,3 +442,185 @@ func TestDBAuthService_Authorize(t *testing.T) {
...
+			}
+			gotGroupNames := make([]string, 0, len(testCase.groupNames))
+			groups, _, err := s.ListGroups(&model.PaginationParams{Amount: -1})
+			if err != nil {
+				t.Errorf("ListGroups: %s", err)
+			}
+			for _, group := range groups {
+				gotGroupNames = append(gotGroupNames, group.DisplayName)
+			}
+			sort.Strings(testCase.groupNames)
```
Comment (johnnyaug) [inline auth/service_test.go:552]: same here with the sorting

File: auth/service_test.go [hunk 3]
```diff
@@ -379,3 +442,185 @@ func TestDBAuthService_Authorize(t *testing.T) {
...
+	// and Postgres time storage.
+	time := time.Date(2222, 2, 22, 22, 22, 22, 0, time.UTC)
+	if err := s.CreateUser(&model.User{DisplayName: userName, ID: -22, CreatedAt: time}); err != nil {
+		t.Fatalf("CreateUser(%s): %s", userName, err)
+	}
+	user, err := s.GetUser(userName)
+	if err != nil {
+		t.Fatalf("GetUser(%s): %s", userName, err)
+	}
+	userById, err := s.GetUserByID(user.ID)
```
Comment (johnnyaug) [inline auth/service_test.go:598]: userById is a bit misleading as a name, sounds like a map

**Review by johnnyaug — APPROVED** [review]
Minor comments

File: auth/service_test.go (hunk 3 above)
Comment (arielshaqed) [reply auth/service_test.go]: Changed to `gotUser`.

**Review by arielshaqed — COMMENTED** [review]
Thanks!

**Pull request merged**